src/metabase/driver/common/parameters/values.clj

```diff
@@ -206,9 +219,15 @@
                                            {:type                qp.error-type/invalid-parameter
                                             :template-tag        tag
                                             :matching-parameters params})))
-                         (first matching-params))]
+                         (first matching-params))
+        nil-value?       (and matching-param
+                              (nil? (:value matching-param)))]
     ;; if both the tag and the Dashboard parameter specify a default value, prefer the default value from the tag.
     (or (:value matching-param)
+        (when (and nil-value? (:required tag))
```

Comment:
I'd put a comment here saying why we're doing this. Something along the line: if the param is present and the value is nil, this is a way to bypass the default value of a parameter.

Parameters is complicated enough and it'd benefit a lot for future refactors if we document the desired behaviors here.